src/MementoContainer/Analysis/PropertyAnalyzer.cs

```diff
@@ -128,11 +149,31 @@ public IList<IPropertyData> GetProperties(object obj)
         }
 
         /// <summary>
+        /// Tells if a given property has to be registered in the memento in case the class is decorated with MementoClassAttribute
+        /// </summary>
+        /// <param name="kv">Property info and its attributes</param>
+        /// <returns>True if the property must be registered in the memento</returns>
+        protected virtual bool PropertySelectorForMementoClass(KeyValuePair<PropertyInfo, IList<Attribute>> kv)
+        {
+            return kv.Value.Any(attr => attr is MementoCollectionAttribute) || kv.Key.PropertyType.IsCollection();
+        }
+
+        /// <summary>
+        /// Tells if a given property has to be registered in the memento in case the class is not decorated with MementoClassAttribute
+        /// </summary>
+        /// <param name="kv">Property info and its attributes</param>
+        /// <returns>True if the property must be registered in the memento</returns>
+        protected virtual bool PropertySelectorForNonMementoClass(KeyValuePair<PropertyInfo, IList<Attribute>> kv)
+        {
+            return kv.Value.Any(attr => attr is MementoCollectionAttribute);
```

Comment:
Was this a copy-paste oversight? I think this is meant to be

```csharp
kv.Value.Any(attr => attr is MementoPropertyAttribute)
```

Reply:
Yes, the same as the last one 😄